src/main/java/org/amahi/anywhere/adapter/ServerFilesAdapter.java

```diff
@@ -122,61 +124,74 @@ public void setUpDownloadReceiver() {
     public void onBindViewHolder(RecyclerView.ViewHolder holder, final int position) {
         final ServerFileViewHolder fileHolder = (ServerFileViewHolder) holder;
 
-        final ServerFile file = filteredFiles.get(position);
+        if (showShimmer) {
+            fileHolder.shimmerFrameLayout.startShimmer();
+        } else {
+            final ServerFile file = filteredFiles.get(position);
 
-        if (Mimes.match(file.getMime()) == Mimes.Type.DIRECTORY) {
-            fileHolder.moreInfo.setVisibility(View.GONE);
-            fileHolder.moreOptions.setVisibility(View.GONE);
-            fileHolder.rightArrow.setVisibility(View.VISIBLE);
+            fileHolder.shimmerFrameLayout.stopShimmer();
+            fileHolder.shimmerFrameLayout.setShimmer(null);
+            fileHolder.fileIconView.setBackground(null);
+            fileHolder.fileTextView.setBackground(null);
+            fileHolder.moreOptions.setBackground(null);
+            fileHolder.fileLastModified.setBackground(null);
+            fileHolder.fileSize.setBackground(null);
```

Comment:
Rather call that stop shimmer function only so that it's at one place. Would help in future in case we plan to change something in how shimmer works.

Reply:
I have created a function `stopShimmer` that contains all the code for stopping the shimmer at one place.